src/testdir/test_popup.vim

```diff
@@ -1986,4 +1986,75 @@ func Test_pum_complete_with_special_characters()
   call StopVimInTerminal(buf)
 endfunc
 
+func Test_pum_maxwidth()
+  CheckScreendump
+
+  let lines =<< trim END
+    123456789_123456789_123456789_a
+    123456789_123456789_123456789_b
+                123
+  END
+  call writefile(lines, 'Xtest', 'D')
+  let buf = RunVimInTerminal('Xtest', {})
+
+  call term_sendkeys(buf, "GA\<C-N>")
+  call VerifyScreenDump(buf, 'Test_pum_maxwidth_01', {'rows': 8})
+  call term_sendkeys(buf, "\<Esc>u")
```

Comment:
Can you avoid using `u` in screendump tests? Using `u` is a source of flaky tests when the undo message is include in the screendump. Although that isn't the case here, it's easy to make mistakes.

Reply:
okay there is reference test_popup_position

Reply:
Oh, I didn't notice that the line is initially non-empty here. Then maybe using `u` is fine.